contracts/interchain-token-service/tests/interchain_transfer.rs

```diff
@@ -0,0 +1,130 @@
+use axelar_gateway::testutils::{generate_proof, get_approve_hash};
+use axelar_gateway::types::Message as GatewayMessage;
+use axelar_soroban_std::{assert_emitted_event, assert_last_emitted_event, assert_ok};
+use interchain_token_service::testutils::{
+    bytes_from_hex, register_chains, setup_env, setup_gas_token,
+};
+use interchain_token_service::types::{HubMessage, InterchainTransfer, Message};
+use soroban_sdk::xdr::ToXdr;
+use soroban_sdk::{testutils::Address as _, vec, Address, BytesN, String, Symbol};
+
+const HUB_CHAIN: &str = "hub_chain";
+
+#[test]
+fn message_routing_send() {
+    let (env, client, gateway_client, _) = setup_env();
+    register_chains(&env, &client);
+    let sender: Address = Address::generate(&env);
+    let gas_token = setup_gas_token(&env, &sender);
+
+    let token_id = BytesN::from_array(&env, &[255u8; 32]);
+    let destination_chain = String::from_str(&env, HUB_CHAIN);
+    let destination_address = bytes_from_hex(&env, "4F4495243837681061C4743b74B3eEdf548D56A5");
+    let amount = i128::MAX;
+    let data = Some(bytes_from_hex(&env, "abcd"));
+
+    let msg = Message::InterchainTransfer(InterchainTransfer {
+        token_id: token_id.clone(),
+        source_address: sender.clone().to_xdr(&env),
+        destination_address: destination_address.clone(),
+        amount,
+        data: data.clone(),
+    });
+
+    let expected_payload = assert_ok!(HubMessage::SendToHub {
+        destination_chain: String::from_str(&env, HUB_CHAIN),
+        message: msg.clone()
+    }
+    .abi_encode(&env));
+
+    let expected_destination_chain = client.its_hub_chain_name();
+    let expected_destination_address = client
+        .trusted_address(&client.its_hub_chain_name())
+        .unwrap();
+    let expected_payload_hash: BytesN<32> = env.crypto().keccak256(&expected_payload).into();
+
+    client.interchain_transfer(
+        &sender,
+        &token_id,
+        &destination_chain,
+        &destination_address,
+        &amount,
+        &data,
+        &gas_token,
+    );
+
+    assert_last_emitted_event(
```

Comment:
this should be switched to using the new style event test with goldie that Chris introduced in a follow up PR